### ISSUE TITLE
prune on b2: restic stopped reading files for new index after 36k files

### ISSUE THREAD

**Issue by fd0** [restic/restic#2024]
Output of `restic version`
--------------------------

0.9.2

How did you run restic exactly?
-------------------------------

restic prune

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

B2

Expected behavior
-----------------

Prune successfully

Actual behavior
---------------

```
building new index for repo
[1:28:48] 68.48%  36000 / 52567 packs
repository contains 36000 packs (606127 blobs) with 170.894 GiB
processed 606127 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 8 snapshots
[0:33] 100.00%  8 / 8 snapshots
Fatal: number of used blobs is larger than number of available blobs!
```

Steps to reproduce the behavior
-------------------------------

no idea

Do you have any idea what may have caused this?
-----------------------------------------------

Yep, when `List()` returns an error, it is silently discarded here:

https://github.com/restic/restic/blob/233596f4bcfc6ff7f51aa82e764551cc973f318e/internal/list/list.go#L64

Do you have an idea how to solve the issue?
-------------------------------------------

Pass the error up the call chain and/or retry it.

**Comment by MichaelEischer** [comment]
Shouldn't this be fixed since b096fc7abf7d8a3ef2ab33b6499005dc1f7cd164 ?

**Comment by fd0** [comment]
Likely, I'll close this issue for now.
